MyLibrary/BusinessLogic/LibraryService/GenreService.cs

```diff
@@ -0,0 +1,114 @@
+﻿// <copyright file="GenreService.cs" company="Peretiatko Anastasiia">
+// Copyright (c) Peretiatko Anastasiia. All rights reserved.
+// </copyright>
+
+using BusinessLogic.DataProvider;
+using BusinessLogic.LibraryModel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace BusinessLogic.LibraryService
+{
+    /// <summary>
+    /// Genre services
+    /// </summary>
+    /// <seealso cref="BusinessLogic.LibraryService.LibraryService" />
+    /// <seealso cref="BusinessLogic.LibraryService.IGenreService" />
+    public class GenreService : LibraryService, IGenreService
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GenreService"/> class
+        /// </summary>
+        /// <param name="dataProvider">The data provider</param>
+        public GenreService(IDataProvider dataProvider) : base(dataProvider)
+        {
+        }
+
+        #region GenreServices
+
+        /// <summary>
+        /// Gets all genres
+        /// </summary>
+        /// <returns>The list of genres in Library</returns>
+        public IEnumerable<Genre> GetAll()
+        {
+            return this.genres;
+        }
+
+        /// <summary>
+        /// Gets the genre by identifier
+        /// </summary>
+        /// <param name="id">The genre identifier</param>
+        /// <returns>The found genre or null if there is no such genre in library</returns>
+        public Genre GetById(uint id)
+        {
+            return this.genres.FirstOrDefault(genre => genre.Id == id);
+        }
+
+        /// <summary>
+        /// Adds the genre to Library
+        /// </summary>
+        /// <param name="genre">The genre for adding</param>
+        /// <returns>The id of added genre</returns>
+        /// <exception cref="System.ArgumentException">The genre with such id already exists!</exception>
+        public uint Add(Genre genre)
+        {
+            if (this.genres.Any(someGenre => someGenre.Id == genre.Id))
+            {
+                throw new ArgumentException("The genre with such id already exists!");
+            }
+
+            this.genres.Add(genre);
+            return genre.Id;
+        }
+
+        /// <summary>
+        /// Updates the genre
+        /// </summary>
+        /// <param name="genreId">The genre identifier</param>
+        /// <param name="genre">The new genre</param>
+        /// <returns>The id of updated genre or null if there is no genre with specified id</returns>
+        public uint? Update(uint genreId, Genre genre)
+        {
+            Genre genreForUpdate = this.genres.FirstOrDefault(someGenre => someGenre.Id == genreId);
```

Comment:
someGenre is strange name - use just genre

Reply:
I can`t use genre because it is in my function`s parameters, that`s why I have thought up this naming. Can I use just "x" in my next levels?
Genre genreForUpdate = this.genres.FirstOrDefault(x => x.Id == genreId);

Reply:
Probably “g” will take more sense.
If lamda query is small it is acceptable to minify name to 1-2 characters and common. 
But it is more preferable to make normal names as everywhere else. Because if you will start use 1-2 characters it’s very difficult to change such habits, I can assure you by personal experience